src/pages/RegisterCardNickName.tsx

```diff
@@ -1,29 +1,40 @@
 import { Card } from 'components/common/Card/types';
+import { RegisterCardLoadingBar } from 'components/common/LodaingBar/RegisterCardLoadingBar';
 import { AddCardNickNameForm } from 'components/Form/AddCardNickNameForm';
 import { PageContainer } from 'components/style/PageContainer';
+import { useCardAddForm } from 'contexts/CardAddFormProvider';
 import { useUserCards } from 'contexts/UserCardProvider';
-import { useLocation, useNavigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 import styled from 'styled-components';
 
 export const RegisterCardNickName = () => {
+  const { card, resetCardForm } = useCardAddForm();
+  const [isSubmitting, setIsSubmitting] = useState(false);
   const {
     actions: { addCard },
   } = useUserCards();
   const navigate = useNavigate();
-  const location = useLocation();
 
   const goHome = () => {
     navigate('/');
   };
 
   const handleCardNickNameSubmit = (card: Card) => {
-    addCard(card);
-    goHome();
+    setIsSubmitting((prev) => !prev);
```

Comment:
Submit을 준비하는 코드 아닌가요?
```suggestion
    setIsSubmitting(true);
```